test/it/RetrieveAllLegalUnitsForEnterpriseAcceptanceSpec.scala

```diff
@@ -0,0 +1,90 @@
+          aColumnWith(name = uBRN, value = UBRNOne.value),
+          aColumnWith(name = UBRNref, value = s"one-luref"),
+          aColumnWith(name = ern, value = TargetErn.value)),
+        aRowWith(key = s"${LegalUnitQuery.byRowKey(TargetErn, TargetPeriod, UBRNTwo)}", columns =
+          aColumnWith(name = uBRN, value = UBRNTwo.value),
+          aColumnWith(name = ern, value = TargetErn.value),
+          aColumnWith(name = entref, value = "two-entref"))
+      ).mkString("[", ",", "]")
+    }}"""
+
+  info("As a SBR user")
+  info("I want to retrieve all legal units for an enterprise and a period in time")
+  info("So that I can view the legal unit details via the user interface")
+
+  feature("retrieve Legal Units by exact Enterprise reference (ERN) and period") {
+    scenario("when the enterprise has multiple legal units") { wsClient =>
+      Given(s"two legal units exist with $TargetErn for $TargetPeriod")
+
+      stubHBaseFor(anAllLegalUnitsForEnterpriseRequest(withErn = TargetErn, withPeriod = TargetPeriod).willReturn(
+        anOkResponse().withBody(LegalUnitMultipleMatchHBaseResponseBody)
+      ))
+
+      When(s"the legal units with $TargetErn for $TargetPeriod are requested")
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/legalunits").get())
+
+      Then(s"the details of the two legal units with $TargetErn for $TargetPeriod are returned")
+      response.status shouldBe OK
+      response.header(CONTENT_TYPE).value shouldBe JSON
+      response.json.as[Seq[LegalUnit]] should contain theSameElementsAs Seq(
+        LegalUnit(UBRNOne, UBRNref = Some("one-luref"), enterprise = EnterpriseLink(TargetErn, entref = None)),
+        LegalUnit(UBRNTwo, UBRNref = None, enterprise = EnterpriseLink(TargetErn, entref = Some("two-entref")))
+      )
+    }
+
+    info("An existing enterprise and period in time combination should have at least one legal unit")
+
+    scenario("when there is no such combination of Enterprise reference (ERN) and period") { wsClient =>
+      Given(s"no legal units exist with $TargetErn for $TargetPeriod")
+      stubHBaseFor(anAllLegalUnitsForEnterpriseRequest(withErn = TargetErn, withPeriod = TargetPeriod).willReturn(
+        anOkResponse().withBody(NoMatchFoundResponse)
+      ))
+
+      When(s"the legal units with $TargetErn for $TargetPeriod are requested")
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/legalunits").get())
+
+      Then("a NOT FOUND response is returned")
+      response.status shouldBe NOT_FOUND
+    }
+    scenario("when the specified period is invalid") { wsClient =>
+      Given(s"that a valid Period has yyyyMM format")
+      val invalidPeriod = "2018-02"
+      stubHBaseFor(anAllLegalUnitsForEnterpriseRequest(withErn = TargetErn, withPeriod = TargetPeriod).willReturn(
```

Comment:
Do we need to stub here?  The router should reject the request, and so we should never make a REST call.